be/shopveeAPI/Services/Address/AddressServices.cs

```diff
@@ -0,0 +1,123 @@
+﻿using AutoMapper;
+using Microsoft.EntityFrameworkCore;
+using shopveeAPI.Common;
+using shopveeAPI.DbContext;
+using shopveeAPI.Models;
+using shopveeAPI.Repository;
+using shopveeAPI.Services.Address.Dto.Request;
+using shopveeAPI.Services.Address.Dto.Response;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace shopveeAPI.Services.Address
+{
+    public class AddressServices : GenericRepository<AddressEntity>, IAddressServices
+    {
+        private readonly ShopveeDbContext _shopveeDbContext;
+        private readonly IMapper _mapper;
+
+        private const int MaxAddress = 10;
+
+        public AddressServices(ShopveeDbContext shopveeDbContext, IMapper mapper) : base(shopveeDbContext)
+        {
+            _shopveeDbContext = shopveeDbContext;
+            _mapper = mapper;
+        }
+
+        public async Task<ServiceResponse> Add(AddressRequest request)
+        {
+            try
+            {
+                var listAddressCount = await _shopveeDbContext.Address.CountAsync();
+
+                if (listAddressCount >= MaxAddress)
+                {
+                    return BadRequest("No more addresses allowed.");
+                }
+
+                AddressEntity addressEntity = _mapper.Map<AddressEntity>(request);
+                addressEntity.IsDefault = listAddressCount == 0 || request.IsDefault;
+
+                if (addressEntity.IsDefault)
+                {
+                    await _shopveeDbContext.Address.Where(x => x.IsDefault).ForEachAsync(x => x.IsDefault = false);
+                }
+
+                await _shopveeDbContext.Address.AddAsync(addressEntity);
+                await _shopveeDbContext.SaveChangesAsync();
+
+                var response = _mapper.Map<AddressResponse>(addressEntity);
+                return Ok(response);
+            }
+            catch (Exception e)
+            {
+                return BadRequest(e.Message);
+            }
+        }
+
+        public async Task<ServiceResponse> Update(Guid id, AddressRequest request)
+        {
+            try
+            {
+                if (id != request.Id)
+                {
+                    return NotFound();
+                }
+
+                var newEntity = _mapper.Map<AddressEntity>(request);
+                _shopveeDbContext.Address.Update(newEntity);
+                await _shopveeDbContext.SaveChangesAsync();
+
+                var response = _mapper.Map<AddressResponse>(newEntity);
+                return Ok(response);
+            }
+            catch (Exception e)
+            {
+                return BadRequest(e.Message);
+            }
+        }
+
+        public async Task<ServiceResponse> GetAllAddressAsync()
+        {
+            try
+            {
+                var addresses = await _shopveeDbContext.Address.ToListAsync();
```

Comment:
replace `addresses` by `entities`. double check and apply same convention to others